infrastructure/src/main/java/org/corfudb/infrastructure/log/FileSystemAgent.java

```diff
@@ -129,4 +157,46 @@ private void checkLimits() {
             }
         }
     }
+
+    public static class PartitionAgent {
+        @Getter
+        private PartitionAttribute partitionAttribute;
+        private final Path logPartition;
+        private final FileSystemConfig config;
+        public PartitionAgent(FileSystemConfig config) {
+            this.config =  config;
+            logPartition = Paths.get(config.logDir.getRoot().toString(), config.logDir.subpath(0, 1).toString());
+            partitionAttribute = setPartitionAttribute();
+            //we need a scheduler
```

Comment:
Why is a scheduler needed here?

The partition entry can be loaded on every check. What's the point of having it on a different cadence, is it that expensive?

Reply:
Let's discuss it in #3121
Maithem, could you also please explain your concern in more detail in Chetan's pr?